addons/mail/models/mail_thread.py

```diff
@@ -1275,7 +1275,7 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
             message_dict.pop('parent_id', None)
 
             # check it does not directly contact catchall
-            if catchall_alias and any(email.startswith(catchall_alias) for email in email_to_localparts):
+            if catchall_alias and all(email_localpart == catchall_alias for email_localpart in email_to_localparts):
```

Comment:
@MiquelRForgeFlow You nailed this line, as explained in https://github.com/OCA/OCB/pull/916/files#r393558830.